hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/conf/OzoneConfiguration.java

```diff
@@ -242,10 +209,25 @@ public boolean equals(Object obj) {
     }
   }
 
+  /** Add default resources. */
   public static void activate() {
-    // adds the default resources
-    Configuration.addDefaultResource("hdfs-default.xml");
-    Configuration.addDefaultResource("hdfs-site.xml");
+    // core-default and core-site are added by parent class
+    addDefaultResource("hdfs-default.xml");
+    addDefaultResource("hdfs-site.xml");
+
+    // One generated file per module
+    addDefaultResource("hdds-common-default.xml");
```

Comment:
Can we extra the module name into an array or list, and append the "default.xml" automatically for each module, can let a comment telling about if there is any new module added in future, and the new module uses this annotation based property definition mechanism, module name should be added into this array?

Reply:
Will do, waiting a bit for any other feedback.

Reply:
So, now we generate these default xml files per module and load each file once right?

Reply:
Yes.

Reply:
That works, thanks